src/compiler/crystal/compiler.cr

```diff
@@ -295,7 +297,15 @@ module Crystal
 
       Crystal.timing("Codegen (linking)", @stats) do
         Dir.cd(output_dir) do
-          system %(#{CC} -o "#{output_filename}" "${@}" #{@link_flags} #{lib_flags}), object_names
+          if thin_lto
+            lto_cache_dir = "#{output_dir}/lto.cache"
+            Dir.mkdir_p(lto_cache_dir)
+            cc = ENV["CC"]? || "clang -flto=thin -Wl,-mllvm,-threads=#{n_threads},-cache_path_lto,#{lto_cache_dir},#{@release ? "-mllvm,-O2" : "-mllvm,-O0"}"
```

Comment:
Note that with ThinLTO the .bc stored in .o files with augmented metadata. This is different than `--cross-compile`.